src/common/util/game.js

```diff
@@ -257,12 +257,31 @@ export function logAction(state, player, action, cards, timestamp, target = null
   return state;
 }
 
-export function newGame(state, player, usernames, decks, seed = 0) {
-  state = Object.assign(state, cloneDeep(defaultState), {player: player, rng: seededRNG(seed)}); // Reset game state.
-  state.usernames = usernames;
-  state.players.blue = bluePlayerState(decks.blue);
-  state.players.orange = orangePlayerState(decks.orange);
-  state.started = true;
+// gameMode can be 'normal' or 'sharedDeck'
+export function newGame(state, player, usernames, decks, seed = 0, gameMode = DEFAULT_GAME_MODE) {
```

Comment:
Should have a private `newNormalGame` and `newSharedGame`, then `newGame` becomes:

```
initialize state...
if (gameMode === DEFAULT_GAME_MODE) {
    newNormalGame(...);
} else if (gameMode === SHARED_GAME_MODE) {
    newSharedGame(...);
} else {
   error handling
}
```

Which seems a bit simpler.